### PR TITLE
Removing API publishing functionality and referring default security from user namespace 

### PR DESCRIPTION
**Pull request by Shehanir**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: apim-operator/deploy/controller-artifacts/operator.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ spec:
       containers:
         - name: apim-operator
           # Replace this with the built image name
-          image: wso2am/apim-operator:v0.5
+          image: wso2am/apim-operator:v2.7
```
Comment (pubudu538) [inline apim-operator/deploy/controller-artifacts/operator.yaml]: Why a sudden increase? Let's keep v1.0 for the initial release.

Comment (Shehanir) [reply apim-operator/deploy/controller-artifacts/operator.yaml]: changes are added

File: apim-operator/pkg/controller/api/api_controller.go [hunk 2]
```diff
@@ -244,8 +253,53 @@ func (r *ReconcileAPI) Reconcile(request reconcile.Request) (reconcile.Result, e
...
+
+		if errGetSec != nil && errors.IsNotFound(errGetSec) {
+			var defaultCertName string
+			var defaultCertvalue []byte
+			//retrieve default-security from wso2-system namespace
+			errSec := r.client.Get(context.TODO(), types.NamespacedName{Name:  defaultSecurity, Namespace: wso2NameSpaceConst}, securityDefault)
+			if errSec != nil && errors.IsNotFound(errSec){
+				reqLogger.Info("default security instance is not found in wso2-system namespace")
+				return reconcile.Result{}, errSec
+			}else if errSec != nil{
```
[the remark this answers is not in the document]
Comment (Shehanir) [reply apim-operator/pkg/controller/api/api_controller.go:268]: This is already fixed in the latest code

File: apim-operator/pkg/controller/api/api_controller.go [hunk 3]
```diff
@@ -244,8 +253,53 @@ func (r *ReconcileAPI) Reconcile(request reconcile.Request) (reconcile.Result, e
...
+			}
+			//copying default security to user namespace
+			newDefaultSecurity := copyDefaultSecurity(securityDefault, userNameSpace)
+			errCreateSecurity := r.client.Create(context.TODO(),newDefaultSecurity)
+			if errCreateSecurity != nil {
+				log.Error(errCreateSecurity,"error creating secret for default security in user namespace")
+				return reconcile.Result{},errCreateSecurity
+			}
+		}
+
```
[the remark this answers is not in the document]
Comment (Shehanir) [reply apim-operator/pkg/controller/api/api_controller.go:300]: This is already fixed in the latest code

**Pull request merged**